File: NetBeansProjects/java/payroll/src/main/java/com/maicol/payroll/Supervisor.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.maicol.payroll;

/**
 *
 * @author Maicol
 */


/**
 *
 * @author Maicol
 */
   public class Supervisor extends person {
    private final int bonus;
    private final int totalBonus;
    private final Coach coach;

    public Supervisor(Coach coach, int bonus) {
        super("Julian Esteban", "Diaz", 27, 85000, (27 * 85000 < 2000000) ? 200000 : 0, 27 * 85000, "Supervisor");
        this.coach = coach;
        this.bonus = bonus;
        this.totalBonus = (coach.getCustomersAttended() / 5) * bonus;
    }

    @Override
    public double calculateSalary() {
        return (getDaysWorked() * getDayValue()) - totalBonus;
    }

    public int getTotalBonus() {
        return totalBonus;
    }

    @Override
    public void show() {
        System.out.println("Information of the worker: " + getPosition());
        System.out.println("Position: " + getPosition());
        System.out.println("Names and last names: " + getNames() + " " + getLastNames());
        System.out.println("Days worked: " + getDaysWorked());
        System.out.println("Day value: " + getDayValue() + "$");
        System.out.println("With a salary of: " + getSalary() + "$");
        System.out.println("With a transport allowance of: " + getTransport() + "$");
        System.out.println("Gaining a bonus of: " + getTotalBonus() + "$");
        System.out.println("For obtaining a full payment of: " + calculateSalary() + "$");
    }
}

File: php/examples/payroll/php/supervisor.php
<?php
include_once('coach.php');
include_once('deductions.php');
    class supervisor extends person{
    use deductions;

        private int $bonus;
        private int $totalBonus=0;
        private coach $coach;

        public function __construct(coach $coach, int $bonus, int $totalBonus=0){
            $this->coach=$coach;
            $trainedCustomers=$this->coach->getCustomersAttented();
            $this->bonus=$bonus;
            $this->totalBonus=intdiv($trainedCustomers,5)*$this->bonus;
            $daysWorked=27;
            $dayValue=85000;
            $salary=$daysWorked*$dayValue;
            $transport= ($salary<2000000)?200000:0;
        
        parent:: __construct("Julian Esteban","Diaz",$daysWorked,$dayValue, $transport,$salary,"Supervisor"); 
        $this-> initDeductions($salary);
        }
            public function calculateSalary():float{
                 return ($this->getDaysWorked()*$this->getDayValue())-$this->totalDeductions;
            }
            public function getTotalBonus():int{
                return $this->totalBonus;
            }
            public function show():void{
                echo "<h1>Information of the worker: ".$this->getPosition()."</h1>";
                echo "<h2>Position: ".$this->getPosition(). "</h2>";
                echo "<p>Names and last names: " . $this->getNames() ." ". $this->getLastNames() ."</p>";
                echo "<p>Days worked: "  .$this->getDaysWorked(). "</p>";
                echo "<p>Day value: " .$this->getDayValue()."$"."</p>";
                echo "<p>With a salary of: ". $this->getSalary()."$"."</p>";
                echo "<p>With a auxiliary of transport of: ". $this->getTransport()."$". "</p>";
                echo "<p>Gaining a bonus of: " . $this->getTotalBonus(). "$". "</p>";
                $this->showDeductions();
                echo "<p>For obtaing a fullpayment of: ". $this->calculateSalary()."$"."</p>";
            }            
    }
?>